components/codec/src/error.rs

```diff
@@ -50,6 +63,18 @@ impl Fail for Box<ErrorInner> {
 #[fail(display = "{}", _0)]
 pub struct Error(#[fail(cause)] pub Box<ErrorInner>);
 
+impl Error {
+    pub fn maybe_clone(&self) -> Option<Error> {
+        self.0.maybe_clone().map(Error::from)
+    }
+}
+
+impl std::error::Error for Error {
```

Comment:
`Error` already derives `Fail`, you can use `Fail` infra and no need to implement `std::Error` for it again.